### ISSUE TITLE
[VL] Some Memory is not controlled by MemoryAllocator

### ISSUE THREAD

**Issue by jackylee-ch** [oap-project/gluten#1194]
**Describe the bug**
While we are testing TPCDS, we found the memory used by executor is bigger than configured (memory + memoryOverhead + offHeap). This problem is particularly evident in the test of q95.

**To Reproduce**
Run TPCDS queries and use `top` to show executor memory.

**Comment by zhouyuan** [comment]
@zhztheplayer

**Comment by zhztheplayer** [comment]
@jackylee-ch Are you working on this? Or I'm going to submit a fix to include the mmap allocation into Gluten memory management

**Comment by jackylee-ch** [comment]
Sorry, I don't have it. The reason why we found this problem is that our scheduling system would detect whether the actual usage of memory exceeds configured, and kill it if exceeds.

**Comment by jinchengchenghh** [comment]
We revoke `GetDefaultWrappedArrowMemoryPool` sometimes in code, if use this one, it will not track by memory allocator.
We should have a memory pool which lifecycle is same with task memory allocator.

Not track scene 

- https://github.com/oap-project/gluten/issues/1108
- https://github.com/oap-project/gluten/blob/main/cpp/core/jni/JniWrapper.cc#L144

**Comment by weiting-chen** [comment]
@jackylee-ch could you help to re-check if q95 case is related to #1228 mmap allocator or there are other cases we have to cover like @jinchengchenghh mentioned?

**Comment by yulongz** [comment]
spark 3.3.2，branch-0.5.0 , centos7.6，tpcds 2T dataset , q95 case has memory allocator error.

spark-sql parameters
```
spark.memory.offHeap.enabled=true
spark.memory.offHeap.size=12g
spark.master=yarn
spark.executor.cores=5
spark.executor.memory=8g
spark.executor.instances=45
spark.sql.shuffle.partitions=480
spark.driver.cores=2
spark.driver.memory=8g
spark.deploy-mode=cluster
spark.plugins=io.glutenproject.GlutenPlugin
spark.gluten.sql.columnar.backend.lib=velox
spark.gluten.loadLibFromJar=true
spark.gluten.sql.columnar.forceshuffledhashjoin=true
spark.shuffle.manager=org.apache.spark.shuffle.sort.ColumnarShuffleManager
spark.files.useFetchCache=false
spark.sql.hive.filesourcePartitionFileCacheSize=0
```

ERROR log 
```
Using Spark's default log4j profile: org/apache/spark/log4j2-defaults.properties
WARNING: Logging before InitGoogleLogging() is written to STDERR
W20230529 15:44:16.903386 3120542 MemoryAllocator.cpp:296] [MEM] Failed to reserve 603979776 bytes for contiguous allocation of 147456 pages, then release 0 bytes from the old allocations
W20230529 15:44:17.901835 3120548 MemoryAllocator.cpp:296] [MEM] Failed to reserve 603979776 bytes for contiguous allocation of 147456 pages, then release 0 bytes from the old allocations
W20230529 15:44:18.618010 3120546 MemoryAllocator.cpp:296] [MEM] Failed to reserve 603979776 bytes for contiguous allocation of 147456 pages, then release 0 bytes from the old allocations
E20230529 15:44:20.605859 3120540 Exceptions.h:68] Line: /opt/gluten/cpp/velox/memory/VeloxMemoryPool.cc:62, Function:~WrappedVeloxMemoryPool, Expression: 0 == remainingBytes (0 vs. 393216) Memory pool op.32.0.0.PartialAggregation should be destroyed only after all allocated memory has been freed. Remaining bytes allocated: 393216, cumulative bytes allocated: 4194304, number of allocations: 13, Source: RUNTIME, ErrorCode: INVALID_STATE
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: (0 vs. 393216) Memory pool op.32.0.0.PartialAggregation should be destroyed only after all allocated memory has been freed. Remaining bytes allocated: 393216, cumulative bytes allocated: 4194304, number of allocations: 13
Retriable: False
Expression: 0 == remainingBytes
Function: ~WrappedVeloxMemoryPool
File: /opt/gluten/cpp/velox/memory/VeloxMemoryPool.cc
Line: 62
Stack trace:
# 0  _ZN8facebook5velox7process10StackTraceC1Ei
# 1  _ZN8facebook5velox14VeloxExceptionC1EPKcmS3_St17basic_string_viewIcSt11char_traitsIcEES7_S7_S7_bNS1_4TypeES7_
# 2  _ZN8facebook5velox6detail14veloxCheckFailINS0_17VeloxRuntimeErrorERKSsEEvRKNS1_18VeloxCheckFailArgsET0_
# 3  _ZNSt23_Sp_counted_ptr_inplaceIN6gluten22WrappedVeloxMemoryPoolESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv
# 4  _ZN8facebook5velox4exec4TaskD2Ev
# 5  _ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE10_M_releaseEv
# 6  _ZN6gluten24WholeStageResultIteratorD1Ev
# 7  _ZNSt19_Sp_counted_deleterIPN6gluten35WholeStageResultIteratorMiddleStageESt14default_deleteIS1_ESaIvELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv
# 8  _ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE10_M_releaseEv
# 9  _ZN5arrow8IteratorISt10shared_ptrIN6gluten13ColumnarBatchEEE6DeleteINS2_14ResultIterator7WrapperINS2_35WholeStageResultIteratorMiddleStageEEEEEvPv
# 10 _ZNSt23_Sp_counted_ptr_inplaceIN6gluten14ResultIteratorESaIS1_ELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv
# 11 _ZNSt8__detail16_Hashtable_allocISaINS_10_Hash_nodeISt4pairIKlSt10shared_ptrIN6gluten14ResultIteratorEEELb0EEEEE18_M_deallocate_nodeEPS9_.isra.0
# 12 Java_io_glutenproject_vectorized_ArrowOutIterator_nativeClose
# 13 0x00007f9e749bd6c7
```

**Comment by jackylee-ch** [comment]
> spark 3.3.2，branch-0.5.0 , centos7.6，tpcds 2T dataset , q95 case has memory allocator error.

@yulongz  Could you retry with main branch? This should already be fixed in main.

**Comment by jackylee-ch** [comment]
As this issue have been fixed by #1228 , we closed it now.
